packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -94,8 +94,10 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
         .getAuthProvider(APIAuthorizationType.iam.authProviderToken);
 
     if (authProvider == null) {
-      throw const AnalyticsException(
-        'No AWSIamAmplifyAuthProvider available. Is Auth category added and configured?',
+      throw const MissingAuthException(
```

Comment:
```suggestion
      throw ConfigurationError(
```

Reply:
I was also thinking that this could be `ConfigurationError`, but also this may be caused by that developer has not added the auth plugin, which is not related to Amplify configuration specifically, and prompts a different action item from checking configuration file and may deserve a dedicated exception, wdyt?